libs/libc/machine/risc-v/common/arch_elf.c

```diff
@@ -343,7 +343,7 @@ int up_relocateadd(FAR const Elf_Rela *rel, FAR const Elf_Sym *sym,
                 addr, _get_val((uint16_t *)addr),
                 sym, sym->st_value);
 
-          offset = (long)sym->st_value - (long)addr;
```

Comment:
should we change offset type to uintptr_t, and remove all (long) castt?

Reply:
But since there is a subtraction here maybe we can use `intptr_t` and not `uintptr_t` just to handle possible errors.

Reply:
I kept the original format on the already existing parts, just fixing the issue with r_addend. 

If you want I can unify the types to the stdint types for the entire file.

Reply:
Ok, let's move it to new patch if you think it's good suggestion since the unification of rv32 and rv64 into the same code base already do many similar change.